### PR TITLE
html-forms - Lawrence Wang

### PR DESCRIPTION
**Pull request by wang-lawrence**
Screen Shot 2023-05-15 at 3 12 44 PM

### CODE REVIEW

File: html-forms/index.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    <title>HTML Forms</title>
+  </head>
+  <body>
+    <h1>Coding Form!</h1>
+
+    <form id = 'coding-form'>
+     <div>Email:</div>
+     <label for="email">
+      <input type="email" id="email" name="user_email">
+     </label>
```
Comment (shawnkost) [inline html-forms/index.html:13]: If you're going to put the `input` inside the `label`, then the `for` attribute isn't needed. It's either use the `for` attribute OR nest the `input` inside the `label`

File: html-forms/index.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+     <div>Why is coding cool?</div>
+     <label for="reason">
+      <select id="reason">
+        <option value='1'>Select an option</option>
+        <option value='2'>Because it's interesting</option>
+        <option value='3'>Because it's fun</option>
+        <option value='4'>Because it's powerful</option>
+      </select>
+     </label>
```
Comment (shawnkost) [inline html-forms/index.html:23]: Same note as above ^

File: html-forms/index.html (hunk 1 above)
Comment (shawnkost) [inline html-forms/index.html:13]: `label` elements need to have some text associated with it. Your text inside your `div` on line 10, should go inside this `label`

File: html-forms/index.html (hunk 2 above)
Comment (shawnkost) [inline html-forms/index.html:23]: `label` needs to have some text associated with it. Utilize your text content inside your `div` above this

File: html-forms/index.html [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      <p>
+        <label for="HTML">
+          <input type="radio" id="HTML" name="language"> HTML
+        </label>
+        <label for="CSS">
+          <input type="radio" id="CSS" name="language"> CSS
+        </label>
+        <label for="JS">
+          <input type="radio" id="JS" name="language"> JavaScript
+        </label>
```
Comment (shawnkost) [inline html-forms/index.html:35]: If you're going to put the `input` inside the `label`, then the `for` attribute isn't needed. It's either use the `for` attribute OR nest the `input` inside the `label`

File: html-forms/index.html [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+      <p>Would you like to sign up for more information?</p>
+      <label for="signup">
+        <input type="checkbox" id="signup"> Sign me up!
+      </label>
+
+      <div>Comments</div>
+      <label for="comment">
+        <textarea cols="30" rows="8" id="comments" placeholder="Tell us your thoughts!"></textarea>
+      </label>
```
Comment (shawnkost) [inline html-forms/index.html:46]: `label` needs to have some text associated with it. Utilize your text content inside the `div` on line 43

**Review by shawnkost — CHANGES_REQUESTED** [review]
A few changes here Lawrence!

**Comment by wang-lawrence** [comment]
Thanks Shawn. I think going forward I will try to not nest them. On FCC they tend to nest <input> in the <label> elements, but still still had the "for" attribute in the <label>. Is there any benefit to doing that?
image

**Review by shawnkost — APPROVED** [review]
Approved. To answer your question, there's no advantage of doing both at the same time. It's unnecessary, it's either one or the other. Not sure why they do that on their site.

**Pull request merged**